src/components/issues-page/kanban-board.tsx

```diff
@@ -1,18 +1,74 @@
 import React from 'react'
+import { DragDropContext, DropResult } from 'react-beautiful-dnd'
 
 import KanbanColumn from './kanban-column'
+import useCurrentUser from '../../effects/useCurrentUser'
+import { changeStatus } from '../../clients/mockApi'
 
 type KanbanBoardProps = {
-  [k in IssueStatus]: readonly Issue[]
+  siteData: SiteData
+  setSiteData: React.Dispatch<React.SetStateAction<SiteData | null>>
 }
 
-const KanbanBoard: React.FC<KanbanBoardProps> = ({ opened, acknowledged, closed }) => {
+const KanbanBoard: React.FC<KanbanBoardProps> = ({ siteData, setSiteData }) => {
+  const currentUser = useCurrentUser()
+
+  const onDragEnd = (result: DropResult) => {
+    const { destination, source } = result
+
+    // Check if the user dragged the item outside of any draggable zone
+    if (!destination) {
+      return
+    }
+
+    // Check if the user dragged the item back to the same spot
+    if (destination.droppableId === source.droppableId && destination.index === source.index) {
+      return
+    }
+
+    if (!currentUser.loaded) {
+      throw new Error(`Cannot update an issue with a nonexistent user`)
+    }
+
+    const sourceStatus = source.droppableId as IssueStatus
+    const destinationStatus = destination.droppableId as IssueStatus
+    const issue: Issue = siteData.issues[sourceStatus][source.index]
+
+    // tslint:disable-next-line: no-expression-statement
+    changeStatus({
+      site: issue.site,
+      id: issue.id,
+      user: currentUser.user,
+      status: destinationStatus,
+      comment: { html: `<div>Changed status from ${sourceStatus} to ${destinationStatus}</div>` },
+    })
+
+    const newIssues = {
```

Comment:
`cloneDeep` works here.

I tried writing a non-mutative approach here, but it was harder to read than what you have. We can eliminate some of the tslint noise by just marking the begin and end of where we should elide the linter

```typescript
    const newIssues = cloneDeep(siteData.issues)

    // Removing the issue from the source column and inserting it to the destination column
    // tslint:disable:no-expression-statement
    newIssues[sourceStatus].splice(source.index, 1)
    newIssues[destinationStatus].splice(destination.index, 0, issue)

    setSiteData({
      ...siteData,
      issues: newIssues,
    })
    // tslint:enable:no-expression-statement
```